tools/backups2datalad/adataset.py

```diff
@@ -268,15 +270,16 @@ async def get_stats(
         files = 0
         size = 0
         substats: dict[str, DatasetStats] = cache if cache is not None else {}
+        # get them all and remap per path
+        subdatasets = {s["path"]: s for s in await self.get_subdatasets()}
         for filestat in await self.get_file_stats():
             path = Path(filestat.path)
             if not is_meta_file(path.parts[0], dandiset=True):
                 if filestat.type is ObjectType.COMMIT:
                     # this zarr should not be present locally as a submodule
-                    # so we should get its id from its information in submodules
-                    sub_info = await self.get_subdatasets(path=path)
-                    assert len(sub_info) == 1  # must be known
-                    zarr_id = Path(sub_info[0]["gitmodule_url"]).name
+                    # so we should get its id from its information in submodules.
+                    sub_info = subdatasets[str(path.absolute())]
```

Comment:
so we need smth like
```suggestion
                    sub_info = subdatasets[str(self.pathobj / path)]
```
?

Reply:
@yarikoptic Yes.